transports/ftp/src/test/java/org/mule/transport/ftp/reliability/FtpRedeliveryPolicyTestCase.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.transport.ftp.reliability;
+
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.junit.Assert.assertThat;
+import org.mule.api.MuleMessage;
+import org.mule.tck.listener.ExceptionListener;
+import org.mule.transport.ftp.AbstractFtpServerTestCase;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import org.junit.Test;
+import org.junit.runners.Parameterized;
+
+public class FtpRedeliveryPolicyTestCase extends AbstractFtpServerTestCase
+{
+
+    private static final String FILE_TXT = "file.txt";
+
+    public FtpRedeliveryPolicyTestCase(ConfigVariant variant, String configResources)
+    {
+        super(variant, configResources);
+    }
+
+    @Parameterized.Parameters
+    public static Collection<Object[]> parameters()
+    {
+        return Arrays.asList(new Object[][] {
+                {ConfigVariant.FLOW, "reliability/ftp-redelivery-policy-config.xml"}
+        });
+    }
+    @Test
+    public void testRedeliveryPolicyDLQConsumesMessage() throws Exception
+    {
+        ExceptionListener exceptionListener = new ExceptionListener(muleContext).setNumberOfExecutionsRequired(3);
```

Comment:
I'd be great if instead of hardcoding a 3 here and a 2 in the flow, you could define a MAX_REDELIVERY_ATTEMPTS constant and expose it as a system property. So, in here you would use MAX_REDELIVERY_ATTEMPS +1 and in the flow ${MAX_REDELIVERY_ATTEMPS}. In this way, you don't spread magic numbers across both the test class and the configuration.

Something like this is implemented in AbstractSftpRedeliveryTestCase